homeassistant/components/portainer/sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Sensor platform for Portainer integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from pyportainer.models.docker import DockerContainer
+
+from homeassistant.components.sensor import (
+    SensorEntity,
+    SensorEntityDescription,
+    StateType,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .coordinator import PortainerConfigEntry, PortainerCoordinator
+from .entity import PortainerContainerEntity, PortainerCoordinatorData
+
+
+@dataclass(frozen=True, kw_only=True)
+class PortainerSensorEntityDescription(SensorEntityDescription):
+    """Class to hold Portainer sensor description.
+
+    value_fn must return a StateType (str | int | float | None or date/datetime/Decimal),
+    but for these container sensors we only return str | None.
+    """
+
+    value_fn: Callable[[DockerContainer], StateType]
```

Comment:
```suggestion
    value_fn: Callable[[DockerContainer], str | None]
```
Always better to be specific